src/QAT/qat/purr/backends/live.py

```diff
@@ -218,20 +222,9 @@ def _execute_on_hardware(self, sweep_iterator: SweepIterator, package: QatFile):
                         aq.mode == AcquireMode.INTEGRATOR
                     )
 
-            playback_results = self.model.control_hardware.start_playback(
```

Comment:
I'm going to need to see an in-depth breakdown on why this is safe considering you're pretty drastically changing the logic. I would also wager the speed increase is minute.

Reply:
So the structure was like:

```
Loop 1: [for aqs in aq_map.values()]

Call + assignment: [playback_results = self.model.control_hardware.start_playback(repetitions=repetitions, repetition_time=repetition_time)]

Loop 2: for [aqs in aq_map.values()]
```

Two changes happened here:

1. Interchange `Loop 1` and `Call + assignment` so it becomes
```
Call + assignment: [playback_results = self.model.control_hardware.start_playback(repetitions=repetitions, repetition_time=repetition_time)]

Loop 1: [for aqs in aq_map.values()]

Loop 2: for [aqs in aq_map.values()]
```

This change is safe because the call `control_hardware.start_playback` invokes the control hardware while `Loop 1` just iterates through `aq_map` and writes to `SimpleAcquire.physical_channel`: both statements target completely different things from different layers and their side effects are isolated from one another.

2. Merge loops `Loop 1` and `Loop 2`
`Loop 1` and `Loop 2` are merged so it becomes:

Call + assignment: [playback_results = self.model.control_hardware.start_playback(repetitions=repetitions, repetition_time=repetition_time)]

Loop 3: [for aqs in aq_map.values()]
```

From a data dependence analysis, each iteration in `Loop 1` reads aq.mode and reads and writes to aq.physical_channel (+ trivial R/W operations), each iteration of `Loop 2` reads fields from `aq` (+ trivial R/W operations). Most importantly, the elements of `acq_map.values()` as well as the elements of `acqs` are independent of one another which makes the R/W effects of `Loop 1` + `Loop 2` linearly constructible and the proposed change is safe.